tests/integration/test_widgets.py

```diff
@@ -0,0 +1,31 @@
+import pytest
+
+
+@pytest.mark.usefixtures("client_class", "empty_test_db")
+class TestWidgets:
+    def setup(self):
+        self.password = "strongestpasswordever"
+
+    def test_get_widgets(self, create_admin_user):
+        user = create_admin_user(pw=self.password)
+        response = self.client.post(
+            "/api/login",
+            json={"email": user.email, "password": self.password},
+        )
+        header = {"Authorization": f"Bearer {response.json['access_token']}"}
+        widgetResponse = self.client.get("/api/widgets", headers=header)
```

Comment:
Thanks for writing tests! Let's simplify this as login is already well tested, and it will make the test a bit easier to read.
```suggestion
    def test_get_widgets(self, valid_header):
        widgetResponse = self.client.get("/api/widgets", headers=valid_header)
```